google/cloud/ndb/model.py

```diff
@@ -604,6 +604,12 @@ def new_entity(key):
 
                 continue
 
+        if value is not None and isinstance(prop, BlobProperty):
+            if name in ds_entity._meanings:
+                meaning, value = ds_entity._meanings[name]
+                if meaning == _MEANING_URI_COMPRESSED:
+                    prop._compressed = True
```

Comment:
This looks suspicious to me. The problem we're trying to solve here is that the software has defined ``_compressed`` for this Property, but it might be stored differently in Datastore. In fact, some entities might have stored this property compressed and some might have stored it uncompressed. We want to read it back out correctly, regardless. But changing the ``_compressed`` flag changes it for *all* entities, not just *this* entity. It also impacts how the property gets written back out, which should probably match what's defined in Python, regardless of how it's been written in the past. 

I *think* the move here is to ignore ``prop._compressed`` and only use ``meaning`` when reading the property in. Likewise, when writing data back out, we should make the meaning (and the data) match the user's definition for ``_compressed``. At no point should we modify ``_compressed``  from what the user has provided.

Reply:
I think you are right. Will take a look.